src/System.Linq/src/System/Linq/Reverse.cs

```diff
@@ -111,6 +111,15 @@ public List<TSource> ToList()
                 return list;
             }
 
+            public HashSet<TSource> ToHashSet(IEqualityComparer<TSource> comparer)
+            {
+                List<TSource> list = _source.ToList();
+
+                list.Reverse();
+
+                return new HashSet<TSource>(list, comparer);
```

Comment:
`new HashSet<TSource>(_source, comparer)` would be the best option here-- `HashSet` is not guaranteed to be ordered and this is a new API, so I don't think there's any harm in doing so. It also leaves us flexibility to change this later if people complain this is unintuitive.

@JonHanna Do you have an opinion here?